pkg/apis/certmanager/v1/types_certificaterequest.go

```diff
@@ -147,7 +191,19 @@ type CertificateRequestStatus struct {
 
 	// The PEM encoded x509 certificate of the signer, also known as the CA
 	// (Certificate Authority).
-	// This is set on a best-effort basis by different issuers.
+	//
+	// The value of this field is based on a best-effort guess made by the issuer:
+	// In case the issuer has a self-signed certificate that can be used
+	// to verify the issued certificate, we are certain that it is the correct
+	// certificate to be used in trust stores.
+	// In case we do not know the self-signed root of the certificate chain,
+	// we make a best-effort guess by assuming the root-most certificate is
+	// the CA that should be trusted. This guess might be incorrect and should
+	// not be relied upon.
```

Comment:
**question:** I got confused trying to reformulate this.


> a best-effort guess made by the issuer

Are you describing how most "cert-manager issuers" behave? Or are you specifying how issuers should behave?

> we are certain that it is the correct certificate to be used in trust stores.

How does the user know that they are in case (1) or in case (2)?

> best-effort guess made by the issuer

What issuer are we talking about here? Is it a "cert-manager issuer" represented by a custom resource such as Issuer, ClusterIssuer or an external issuer custom resource? Or is it the "upstream" issuer (e.g., Let's Encrypt, Vault, or Google Private CA)?

> we are certain [...] In case we do not know the self-signed root [...] we make a best-effort

I'm not sure who "we" refer to. Is it the "cert-manager issuer" as defined above? Or is it the user of cert-manager? Or someone else?

Reply:
I'll move these changes to a seperate PR.